### PR TITLE
many: clarify/tie down model assertion

### PR DESCRIPTION
**Pull request by pedronis**
As discussed this removes unsupported/postponed/reserved headers and make sure they don't sneak in for now in prepare-image, make store optional, defaulting to the ubuntu store if left out.

### CODE REVIEW

File: asserts/header_checks.go [hunk 1]
```diff
@@ -166,3 +166,38 @@ func checkDigest(headers map[string]interface{}, name string, h crypto.Hash) ([]
 
 	return b, nil
 }
+
+func checkShouldBeString(headers map[string]interface{}, name string) (string, error) {
```
Comment (niemeyer) [inline asserts/header_checks.go]: Can we call this `checkOptionalString` and move it to the top, after the other *String ones?

File: asserts/header_checks.go [hunk 2]
```diff
@@ -166,3 +166,38 @@ func checkDigest(headers map[string]interface{}, name string, h crypto.Hash) ([]
...
+	value, ok := headers[name]
+	if !ok {
+		return "", nil
+	}
+	s, ok := value.(string)
+	if !ok {
+		return "", fmt.Errorf("%q header should be a string", name)
```
Comment (niemeyer) [inline asserts/header_checks.go]: s/should/must/? Seems more in-line with the rest.

File: image/image.go [hunk 3]
```diff
@@ -85,6 +89,13 @@ func decodeModelAssertion(opts *Options) (*asserts.Model, error) {
 	if !ok {
 		return nil, fmt.Errorf("assertion in %q is not a model assertion", fn)
 	}
+
+	for _, rsvd := range reserved {
+		if modela.Header(rsvd) != nil {
+			return nil, fmt.Errorf("model assertion cannot have reserved/unsupported header %q set", rsvd)
```
Comment (niemeyer) [inline image/image.go]: :+1:

**Comment by niemeyer** [comment]
A few trivials and LGTM. Thanks!

File: asserts/device_asserts.go [hunk 4]
```diff
@@ -115,7 +96,7 @@ func checkAuthorityMatchesBrand(a Assertion) error {
 	return nil
 }
 
-var modelMandatory = []string{"core", "architecture", "gadget", "kernel", "store", "class"}
```
Comment (mvo5) [inline asserts/device_asserts.go]: Nice to see that store is optional now :)

File: image/image.go [hunk 5]
```diff
@@ -347,12 +362,7 @@ func makeStore(model *asserts.Model) Store {
 	cfg := store.DefaultConfig()
 	cfg.Architecture = model.Architecture()
 	cfg.Series = model.Series()
-	storeID := model.Store()
-	// XXX: make store optional, so defaulting to empty meaning this
-	if storeID == "canonical" {
-		storeID = ""
-	}
-	cfg.StoreID = storeID
+	cfg.StoreID = model.Store()
```
Comment (mvo5) [inline image/image.go]: Yay!

**Comment by mvo5** [comment]
:+1:

**Pull request merged**